### DISCUSSION TITLE
Can not use my GPG key

### DISCUSSION THREAD

**Discussion by juliazero** [WeblateOrg/weblate#9620] [Q&A] [answered]
Hi I'm using weblate-docker and trying to use my gpg key.

What I did:
1. Weblate generated its key after setting WEBLATE_GPG_IDENTITY.
2. I set the HOME variable to: /app/data/home
3. I imported my key and weblate still uses the one generated.
4. I changed WEBLATE_GPG_IDENTITY to a different email and now I can delete this new key using the gpg command.
5. I deleted all keys using the gpg command.
6. After restoring WEBLATE_GPG_IDENTITY to my email it still uses the key that was generated before setting the HOME variable and it is not visible in gpg --list-keys.
How can I fix this?

**Comment by nijel** [comment]
Did you import *secret* key to Weblate? That is needed to sign the commits.

**Reply by juliazero** [reply]
Yes, I imported secret key.
![obraz](https://github.com/WeblateOrg/weblate/assets/11527277/128f14c9-7604-473a-bd73-eeba4d01b167)
I also set correctly WEBLATE_GPG_IDENTITY in docker-compose.override.yml
And weblate still uses wrong key (I restarted the containers and it did not help).
![obraz](https://github.com/WeblateOrg/weblate/assets/11527277/e227120c-9cef-4595-9e7e-a7936775a85a)

**Reply by juliazero** [reply, marked as answer]
There is a problem with the cache. The deleted imported new key is still visible in the webui. I tried to clear the cache, but there is not much described how to do it. I have deleted all volumes, reconfigured and used the generated gpg key from weblate and it seems to work, so I'm not touching this.

**Reply by nijel** [reply]
That's true, I've added a hint about that to the documentation in 98a10a83d7.
